pkg/client/ecr/ecr.go

```diff
@@ -64,28 +66,36 @@ func (c *Client) Tags(ctx context.Context, host, repo, image string) ([]api.Imag
 		return nil, fmt.Errorf("failed to describe images: %s", err)
 	}
 
-	var tags []api.ImageTag
+	tags := map[string]api.ImageTag{}
 	for _, img := range images.ImageDetails {
+		// Base data shared across tags
+		base := api.ImageTag{
+			SHA:       *img.ImageDigest,
+			Timestamp: *img.ImagePushedAt,
+		}
+
 		// Continue early if no tags available
 		if len(img.ImageTags) == 0 {
-			tags = append(tags, api.ImageTag{
-				SHA:       *img.ImageDigest,
-				Timestamp: *img.ImagePushedAt,
-			})
-
+			tags[base.SHA] = base
 			continue
 		}
 
 		for _, tag := range img.ImageTags {
-			tags = append(tags, api.ImageTag{
-				SHA:       *img.ImageDigest,
-				Timestamp: *img.ImagePushedAt,
-				Tag:       tag,
-			})
+			current := base   // copy the base
+			current.Tag = tag // set tag value
+
+			// Already exists — add as child
+			if parent, exists := tags[tag]; exists {
+				parent.Children = append(parent.Children, &current)
+				tags[tag] = parent
+			} else {
+				// First occurrence — assign as root
+				tags[tag] = current
+			}
```

Comment:
I think this logic is reused across clients, so should be moved to util.go or somewhere to keep it DRY

Reply:
Yea, Thats a good catch! 👍  I wasn't sure how each client would work etc and how much it would be needed so copy/paste but yea, deffo make use of it as a helper!